apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java

```diff
@@ -73,6 +74,16 @@ public ApmServerClient(ReporterConfiguration reporterConfiguration) {
 
     public ApmServerClient(ReporterConfiguration reporterConfiguration, List<URL> serverUrls) {
         this.reporterConfiguration = reporterConfiguration;
+        for (ConfigurationOption configurationOption : reporterConfiguration.getConfigurationOptions()) {
```

Comment:
Better create a method `public ConfigurationOption<List<URL>> getServerUrlsOption()` in `ReporterConfiguration` and then just call `reporterConfiguration.getServerUrlsOption().addChangeListener(...)` here

Reply:
fixed